ehsm_kms_service/quote_manager_apis.js

```diff
@@ -0,0 +1,260 @@
+        const { challenge } = payload
+        if (checkStringParam(challenge, true, CHALLENGE_LENGTH_MAX)) {
+            napi_res = napi_result(action, res, [challenge])
+            napi_res && res.send(napi_res)
+        } else {
+            res.send(_result(400, `challenge cannot be empty, must be string and length not more than ${CHALLENGE_LENGTH_MAX}`))
+            return
+        }
+    } catch (e) {
+        console.info('generateQuote :: ', e)
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * Users are expected already got a valid DCAP format QUOTE. And it could use this API to send it to eHSM-KMS to do a quote verification.
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} payload
+ *          ==> {r}quote(String [1~10924]) : A valid DCAP quote in BASE64 string. eg. 'AwACAAAAAAAHAAwAk5pB∗∗∗'
+ *          ==> {r}nonce(String [1~1024]) : A nonce in BASE64 string. eg. 'bm9uY2U='
+ *          ==> {r}policyId(String [36]) : quote policy ID. eg. '326f2049-56ab-474f-a1c0-516de230****'
+ * @returns {Object}
+ *          ==> {r}result(bool) : The result of quote verification. eg. 'true'
+ *          ==> {r}nonce(bool) : The nonce in BASE64 string. eg. 'bm9uY2U=
+ *          ==> {r}sign(String) : The HAMC sign of result and nonce calculated by the API Key. eg. 'T4DRCEZAPLBbb+d3ObD∗∗∗'
+ */
+const verifyQuote = async (res, appid, payload, DB, action) => {
+    try {
+        const { quote, nonce, policyId } = payload
+        if (!checkStringParam(quote, true, QUOTE_LENGTH_MAX)) {
+            res.send(_result(400, `quote cannot be empty, must be string and length not more than ${QUOTE_LENGTH_MAX}.`))
+            return
+        }
+        if (!checkStringParam(nonce, true, NONCE_LENGTH_MAX)) {
+            res.send(_result(400, `nonce cannot be empty, must be string and length not more than ${NONCE_LENGTH_MAX}.`))
+            return
+        }
+        if (!checkStringParam(policyId, true, UUID_LENGTH)) {
+            res.send(_result(400, `policyId cannot be empty, must be string and length equal ${UUID_LENGTH}.`))
+            return
+        }
+        if (!check_UUID_format(policyId)) {
+            res.send(_result(400, 'policyId format wrong.'))
+            return
+        }
+
+        // query quote policy by policyId
+        const query_quote_policy = {
+            selector: {
+                appid,
+                policyId
+            },
+            fields: ['mrenclave', 'mrsigner'],
+            limit: 1
+        }
+        let quote_policy_res = await DB.partitionedFind('quote_policy', query_quote_policy)
+        if (quote_policy_res.docs.length != 1) {
+            res.send(_result(400, 'Invalid  policyId.'))
+            return
+        }
+        mrenclave = quote_policy_res.docs[0].mrenclave
+        mrsigner = quote_policy_res.docs[0].mrsigner
```

Comment:
do we need to check the length?

Reply:
If line 132 (quote_policy_res.docs.length != 1) is false, then the mr_enclave and mr_signer must be a vaild value.so we don't need to check the length.